pages/form/index.tsx

```diff
@@ -0,0 +1,192 @@
+import React, { FormEvent, useState } from "react";
+import axios from "axios";
+import InputForm from "../../components/registration-form/InputForm";
+
+export type FieldName =
```

Comment:
This would be over thinking it. For something like this, 
```
const fieldName = [`email`, `firstName`, `lastName`, `password`, `passwordConfirm`];

// or you can achieve similar effect with
const fieldName = [`email`, `firstName`, `lastName`, `password`, `passwordConfirm`] as const;
```

Reply:
or turns out this following would be more handy 

```
const fields = {
  email: { type: 'text' },
  firstName: {type: 'text'},
  lastName: { type: 'text'},
  password: {type: 'password'},
  passwordConfifm: {type: 'password'},
};
```